### PR TITLE
feat: add fix method

### PR DESCRIPTION
**Pull request by wraithgar**

### CODE REVIEW

**Comment by wraithgar** [comment]
typo fix commit is in this PR too

**Review by hashtagchris — APPROVED** [review]
Looks good. Left some minor non-blocking comments.

**Review by lukekarrys — APPROVED** [review]

File: test/index.js [hunk 1]
```diff
@@ -225,3 +225,22 @@ t.test('cannot update with no content', async t => {
...
+      scripts: true,
+      repository: 'github.com/npm/test-package',
+    }),
+  })
+  const { content } = await PackageJson.fix(path)
+  t.strictSame(content.name, '@npmcli/test-package')
+  t.strictSame(content.version, '1.0.0')
+  t.strictSame(content.bin, { 'test-package': '@npmcli/test-package' })
+  t.strictSame(content.dependencies, { lodash: '' })
+  t.strictSame(content.repository, { type: 'git', url: 'github.com/npm/test-package' })
```
[the remark this answers is not in the document]
Comment (wraithgar) [reply test/index.js:245]: That's a https://github.com/npm/hosted-git-info/ problem

[the remark this answers is not in the document]
Comment (wraithgar) [reply test/index.js:245]:
```javascript
      if (hosted) {
        r = data.repository.url
          = hosted.getDefaultRepresentation() === 'shortcut' ? hosted.https() : hosted.toString()
      }
```

We rely on that module to know if this was a "shortcut" or not, and change our representation accordingly.

**Pull request merged**